IdentityCore/src/util/NSString+MSIDExtensions.m

```diff
@@ -403,10 +160,80 @@ + (NSString *)randomUrlSafeStringOfSize:(NSUInteger)size
         return nil;
     }
     
-    return [NSString msidBase64UrlEncodeData:data];
+    return [data msidBase64UrlEncodedString];
+}
+
+
++ (NSString *)msidHexStringFromData:(NSData *)data
+{
+    const unsigned char *charBytes = (const unsigned char *)data.bytes;
+    
+    if (!charBytes) return nil;
+    NSUInteger dataLength = data.length;
+    NSMutableString *result = [NSMutableString stringWithCapacity:dataLength];
+    
+    for (int i = 0; i < dataLength; i++)
+    {
+        [result appendFormat:@"%02x", charBytes[i]];
+    }
+    
+    return result;
+}
+
+
+/// <summary>
+/// Base64 URL encode a set of bytes.
+/// </summary>
+/// <remarks>
+/// See RFC 4648, Section 5 plus switch characters 62 and 63 and no padding.
+/// For a good overview of Base64 encoding, see http://en.wikipedia.org/wiki/Base64
+/// This SDK will use rfc7515 and encode without using padding.
+/// See https://tools.ietf.org/html/rfc7515#appendix-C
+/// </remarks>
++ (NSString *)msidBase64UrlEncodedStringFromData:(NSData *)data
+{
+    return [[data base64EncodedStringWithOptions:NSDataBase64EncodingEndLineWithLineFeed] msidStringByRemovingPadding];
 }
 
 
+/*! Generate a www-form-urlencoded string of random data */
++ (NSString *)msidUrlFormEncodedStringFromDictionary:(NSDictionary *)dict
+{
+    __block NSMutableString *encodedString = nil;
+    
+    [dict enumerateKeysAndObjectsUsingBlock: ^(id key, id value, BOOL __unused *stop)
+     {
+         NSString *encodedKey = [[[key description] msidTrimmedString] msidUrlFormEncode];
+         
+         if (!encodedString)
+         {
+             encodedString = [NSMutableString new];
+         }
+         else
+         {
+             [encodedString appendString:@"&"];
+         }
+         
+         [encodedString appendFormat:@"%@", encodedKey];
```

Comment:
Why not use appendString?

Reply:
I agree! Brought this over and didn't change this.